src/helpers/promises.ts

```diff
@@ -37,3 +37,106 @@ export const resolveAfterPending = (): Promise<void> => {
 export const resolveNextTick = (): Promise<void> => {
   return new Promise((resolve) => setTimeout(resolve));
 };
+
+/**
+ * A class to manage timing functions and arbitrary periods of time before
+ * generating a `Server-Timing` header for use in HTTP responses.
+ *
+ * This is a very simple implementation that does not support nested timings or
+ * fractions of a millisecond.
+ */
+export class ServerTiming {
+  #timings: Record<
+    string,
+    { description: string; start?: number; end?: number }
+  > = {};
+
+  /**
+   * Start a timing. Returns a function that, when called, will stop the timing
+   * and add it to the header.
+   */
+  public start(name: string, description?: string): () => void {
+    if (this.#timings[name]) {
```

Comment:
Is this possible that this could happen in a serverful env? (see other comment)

Reply:
@djfarrelly We should be okay - each request that comes in creates its own timer (as opposed to a timer for each handler created), so `this.#timings` here will be unique.